PaintingTrivia.py

```diff
@@ -125,32 +139,34 @@ def guess_the_painter():
             return x
 
         try:
-            chosen_painter = int(request.form['chosen_painter'])
+            quiz = pickle.loads(session['quiz'])
+            chosen_painter_id = int(request.form['chosen_painter'])
+            quiz.process_answer(chosen_painter_id)
+            session['quiz'] = pickle.dumps(quiz)
         except KeyError:
-            chosen_painter = -1
-        key = 'right_guesses' if chosen_painter == session[
-            'selected_painter_id'] else 'wrong_guesses'
-        session[key] += 1
-
-    #http://docs.sqlalchemy.org/en/latest/orm/tutorial.html#using-exists
-    #the instructions below correspond to the following sql statement:
-    # SELECT painter.id AS painter_id, painter.name AS painter_name FROM painter
+            print "Something failed"
```

Comment:
bad. by suppressing the error with a meaningless message "Something failed" you neither gain important information (what exactly failed) nor do you recover from the error correctly. You need to put into try/except block only the stuff that can break (in that case it is getting the chosen_painter from the html) and setting it to a default value "-1" could be ok. please change here and elsewhere